rust/agents/relayer/src/msg/metadata/multisig/base.rs

```diff
@@ -52,40 +55,53 @@ pub trait MultisigIsmMetadataBuilder: AsRef<BaseMetadataBuilder> + Send + Sync {
         validators: &[H256],
         threshold: u8,
         metadata: MultisigMetadata,
-    ) -> Vec<u8> {
-        let build_token = |token: &MetadataToken| match token {
-            MetadataToken::CheckpointRoot => metadata.checkpoint.root.to_fixed_bytes().into(),
-            MetadataToken::CheckpointIndex => metadata.checkpoint.index.to_be_bytes().into(),
-            MetadataToken::CheckpointMailbox => {
-                metadata.checkpoint.mailbox_address.to_fixed_bytes().into()
-            }
-            MetadataToken::MessageId => metadata.message_id.unwrap().to_fixed_bytes().into(),
-            MetadataToken::Threshold => Vec::from([threshold]),
-            MetadataToken::MerkleProof => {
-                let proof_tokens: Vec<Token> = metadata
-                    .proof
-                    .unwrap()
-                    .path
-                    .iter()
-                    .map(|x| Token::FixedBytes(x.to_fixed_bytes().into()))
-                    .collect();
-                ethers::abi::encode(&proof_tokens)
-            }
-            MetadataToken::Validators => {
-                let validator_tokens: Vec<Token> = validators
-                    .iter()
-                    .map(|x| Token::FixedBytes(x.to_fixed_bytes().into()))
-                    .collect();
-                ethers::abi::encode(&[Token::FixedArray(validator_tokens)])
-            }
-            MetadataToken::Signatures => {
-                let ordered_signatures = order_signatures(validators, &metadata.signatures);
-                let threshold_signatures = &ordered_signatures[..threshold as usize];
-                threshold_signatures.concat()
+    ) -> Result<Vec<u8>> {
+        let build_token = |token: &MetadataToken| -> Result<Vec<u8>> {
+            match token {
+                MetadataToken::MerkleRoot => Ok(metadata.checkpoint.root.to_fixed_bytes().into()),
```

Comment:
are these named for consistency with the solidity bit? tbh find these confusing, i'd probably rename:

MerkleRoot => CheckpointMerkleRoot
MerkleIndex => MessageMerkleLeafIndex
CheckpointMerkleTree => CheckpointMerkleTreeHook